Gradio GUI

MNeMoNiCuZ opened the pull request:
![440551798-23a6bf06-3978-491e-be8f-7c9db2f13a39](https://github.com/user-attachments/assets/91965227-8d80-4861-8888-5eb0709d58ac)

### Gradio GUI Features
- Basic parameter configuration
- Preset resolutions for common aspect ratios (square, portrait, landscape)
- Continuous generation mode with random seed iteration
- Dynamic prompt enhancement with wildcard support (`{option1|option2|option3}` syntax, as well as `__filename__` for .txt-file wildcards)

ivanprado commented:
This is cool @MNeMoNiCuZ! Could you change it so that `F-Lite` is the default model instead of `F-Lite-Texture`? it would be also nice to have [AGP support](https://github.com/fal-ai/f-lite/blob/main/f_lite/pipeline.py#L171) like we have in the [Comfy node](https://github.com/fal-ai/f-lite/blob/main/f_lite/comfyui.py#L87).

MNeMoNiCuZ commented:
Absolutely. I'll make both of those adjustments.

Any other suggestions or recommendations? I went minimal, but with some nice core functionality.

I wasn't able to make it interrupt or cancel gracefully, so instead it cancels between each generation if you generate automatically.

I was also not able to get batch mode working.

ivanprado commented:
Great!

It would be cool having some kind of prompt enhancing capability, as the
model is much better with complex prompts. We are using superprompt in the
Comfy workflow (https://huggingface.co/roborovski/superprompt-v1) as it is
simple and lightweight enough, but there might be better options.

El jue, 8 may 2025 a las 12:00, MNeMoNiCuZ ***@***.***>)
escribió:

> *MNeMoNiCuZ* left a comment (fal-ai/f-lite#8)
> <https://github.com/fal-ai/f-lite/pull/8#issuecomment-2862473713>
>
> Absolutely. I'll make both of those adjustments.
>
> Any other suggestions or recommendations? I went minimal, but with some
> nice core functionality.
>
> I wasn't able to make it interrupt or cancel gracefully, so instead it
> cancels between each generation if you generate automatically.
>
> I was also not able to get batch mode working.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fal-ai/f-lite/pull/8#issuecomment-2862473713>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG2V2HNYAX25IKP4VPA7XD25MTKVAVCNFSM6AAAAAB4PTQZWKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNRSGQ3TGNZRGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

MNeMoNiCuZ commented:
![image](https://github.com/user-attachments/assets/868afcfa-4cc3-4dcc-8d57-8fd4cc5fd070)

I've now added AGP support.
I also added the following features:

Prompt enhancing using SuperPrompt.
Options for how to handle prompt enhancing and wildcards when a seed is fixed (can be randomized or static while the seed is fixed, at the user's discretion).
Prompt prefix/suffix that are applied AFTER the prompt enhancing. I noticed that the prompt enhancer does not include image types like photography, oil painting or similar.

The results are now much better.

@ivanprado Do you have any preferences in default settings, or having a negative prompt, prefix/suffix pre-filled in?

Right now, no settings are saved between sessions. Anything can be tweaked as a default value, but I have not exposed it to be "easily" accessible, nor user-configurable. This could be worth doing in the future if the GUI gains some popularity to where it makes sense to spend the time improving and supporting it.

ivanprado commented:
I've tried it and it is great @MNeMoNiCuZ ! I few minor changes and it can be merged:
* Rename `f-lite-gradio-gui.py` to `f_lite_gradio_gui.py` as it is more "pythonic"
* Set default resolution to 1024x1024. The model is worse for resolutions below the megapixel, so this would be a better default.

MNeMoNiCuZ commented:
Good idea, I keep forgetting that using - in python file names is a no-no. uhhh, I mean, a no_no.

I set resolutions to

ivanprado commented:
@MNeMoNiCuZ It was trained in many resolutions, but the most common ones during training were the ones [here](https://huggingface.co/spaces/Freepik/F-Lite/blob/main/app.py#L50-L67)

MNeMoNiCuZ commented:
Great, I've updated and added some of these in.
I didn't want to add too many of the same ratios, so I fiddled a bit to keep a fairly concise list.

ivanprado commented:
That's perfect! Thank you very much for contributing @MNeMoNiCuZ . I'm going to merge it :-)

MNeMoNiCuZ commented:
Nice! Happy to contribute :)

Aaaand of course I noticed the first mistake right away :D

The readme uses the old file-name for how to launch the gui:

python f-lite-gradio-gui.py

python f_lite_gradio_gui.py

ivanprado commented:
No worries. Fix submited here https://github.com/fal-ai/f-lite/pull/14
